api.json

```diff
@@ -0,0 +1,802 @@
+                                    "invalid block level": {
+                                        "$ref": "#/components/examples/invalid-block-level-error"
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "/chain/blocks/head": {
+            "get": {
+                "description": "Returns the last applied block.",
+                "tags": [
+                    "chain"
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Current block retrieved.",
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "$ref": "#/components/schemas/Block"
+                                },
+                                "examples": {
+                                    "block not found": {
+                                        "$ref": "#/components/examples/block-not-found-error"
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "/chain/blocks/genesis": {
+            "get": {
+                "description": "Returns the genesis block of the chain.",
+                "tags": [
+                    "chain"
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Genesis block has been retrieved.",
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "$ref": "#/components/schemas/Block"
+                                },
+                                "examples": {
+                                    "block not found": {
+                                        "$ref": "#/components/examples/block-not-found-error"
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "/consensus": {
+            "post": {
+                "description": "nodes related stuff, should only be used by nodes",
+                "tags": [
+                    "consensus"
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Consensus operation accepted."
+                    }
+                }
+            }
+        },
+        "/operations": {
+            "post": {
+                "description": "Sends an operation.",
+                "tags": [
+                    "operations"
+                ],
+                "requestBody": {
```

Comment:
I am wondering about this payload. It is really annoying (to don't say difficult) to build from client side.
Why can't we have one `PUT/POST` endpoint per kind of transaction?
Something like:
- `POST /operations/transaction`
- `POST /operations/vm_transaction`
- `POST /operations/tezos_withdraw`
?

An other naive question: why do we as for block-level? Needing this information, makes the client call the `/block/level` endpoint for each time he/she wants to post an operation. Hence, for each operation, we need 2 HTTP calls 😕 

Same question for the `hash`s, Deku could calculate them...

And last question: Where is the `signature` field? 🤔

Reply:
We need the block level for security reason, it helps to counter replay attack
I've forgotten the signature, but you are totally right, it will be part of the payload